sound_play/scripts/soundplay_node.py

```diff
@@ -257,6 +257,8 @@ def select_sound(self, data):
                         txtfile.write(data.arg.decode('UTF-8').encode('ISO-8859-15'))
                     except UnicodeEncodeError:
                         txtfile.write(data.arg)
+                    except AttributeError:
+                        txtfile.write(data.arg.encode('ISO-8859-15'))
```

Comment:
can you move this except inside the first try code for non `ISO-8859-15` language .
I got following error when I check it with python3.6 + japanese.
(this japanese characters mean "hello")

```python
In [13]: a = 'こんにちは'

In [14]: a.encode('ISO-8859-15')
---------------------------------------------------------------------------
UnicodeEncodeError                        Traceback (most recent call last)
<ipython-input-14-57a6ae6d7f3f> in <module>()
----> 1 a.encode('ISO-8859-15')

/usr/lib/python3.6/encodings/iso8859_15.py in encode(self, input, errors)
     10 
     11     def encode(self,input,errors='strict'):
---> 12         return codecs.charmap_encode(input,errors,encoding_table)
     13 
     14     def decode(self,input,errors='strict'):

UnicodeEncodeError: 'charmap' codec can't encode characters in position 0-4: character maps to <undefined>
```

Reply:
how about like this?

```python
                    try:
                        if hasattr(data.arg, 'decode'):
                            txtfile.write(data.arg.decode('UTF-8').encode('ISO-8859-15'))
                        else:
                            txtfile.write(data.arg.encode('ISO-8859-15'))
                    except UnicodeEncodeError:
                        txtfile.write(data.arg)
```

Reply:
Thanks for the prompt review.

I'm not sure how to fix the case of languages not compatible with `ISO-8859-15`.
I tried to use the released package `sound_play` in Melodic to see how it works in Python2 but it was not able to play the string you provided. Do you have an example of how to use this code to speak Japanese in Python 2 ?

In Python 3 txtfile.write requires a bytes-like object (or encoded string). So if we just write `data.arg` string in the case of `UnicodeEncodeError` the node crashes with 
```
Exception in callback: a bytes-like object is required, not 'str'
```

The only way I see around that is to encode the string before writing it so that we write a bytes object:
- We can not encode it into `ISO-8859-15` as it's not compatible
- If we encode it in `UTF-8` the sound synthesis fails with:
```
Sound synthesis failed. Is festival installed? Is a festival voice installed? Try running "rosdep satisfy sound_play|sh". Refer to http://wiki.ros.org/sound_play/Troubleshooting
```

Reply:
For japanese, we use `voice_text` which require paid licenses and sdk.
https://github.com/jsk-ros-pkg/jsk_3rdparty/tree/master/3rdparty/voice_text

I can check if it will work on Japanese in our lab because we have that license.